cedar-policy-core/src/ast/name.rs

```diff
@@ -373,6 +386,110 @@ mod vars_test {
     }
 }
 
+/// A newtype which indicates that the contained [`Name`] does not contain
+/// reserved `__cedar`, as specified by RFC 52
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
+#[serde(transparent)]
+#[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
+pub struct UnreservedName(pub(crate) Name);
```

Comment:
Would it make more sense to call this one `Name` and the other one something else?  This type is used in a lot of places

Reply:
I renamed it to `Name` and the original `Name` to `ReservedName`, which is kind of ugly.

Reply:
I agree `ReservedName` is kind of ugly.  What about `UncheckedName`?

Reply:
Working on it.

Reply:
Renamed to `UncheckedName`.